test/serverExplorer.test.ts

```diff
@@ -0,0 +1,185 @@
+import * as chai from 'chai';
+import * as sinon from 'sinon';
+import * as sinonChai from 'sinon-chai';
+import { ServersViewTreeDataProvider } from '../src/serverExplorer';
+import { RSPClient, Protocol } from 'rsp-client';
+import { EventEmitter, window } from 'vscode';
+
+const expect = chai.expect;
+chai.use(sinonChai);
+
+suite('Server explorer', () => {
+    let sandbox: sinon.SinonSandbox;
```

Comment:
You're not getting much use out of the sandbox - the point is to create it before the test suite, and then after every test you can call restore to clean up all fakes at once.
Then you should be able to replace all the sinon calls in test cases with sandbox calls.

You can use the docs for inspiration:
https://sinonjs.org/releases/v6.2.0/sandbox/

Reply:
thanks @jrichter1  :)